zerver/lib/upload.py

```diff
@@ -374,19 +373,15 @@ def check_upload_within_quota(realm: Realm, uploaded_file_size: int) -> None:
         raise RealmUploadQuotaError(_("Upload would exceed your organization's upload quota."))
 
 
-def get_file_info(request: HttpRequest, user_file: UploadedFile) -> Tuple[str, Optional[str]]:
+def get_file_info(user_file: UploadedFile) -> Tuple[str, Optional[str]]:
 
     uploaded_file_name = user_file.name
     assert uploaded_file_name is not None
-    content_type = request.GET.get("mimetype")
+    content_type = user_file.content_type
     if content_type is None:
```

Comment:
`user_file.content_type` [defaults](https://github.com/django/django/blob/4.0.6/django/http/multipartparser.py#L237-L239) to `""` rather than `None`.

Also, this loses extra information like the charset in `text/plain; charset=UTF-8`, which gets moved to the `user_file.content_type_extra` dict.

Reply:
> `user_file.content_type` [defaults](https://github.com/django/django/blob/4.0.6/django/http/multipartparser.py#L237-L239) to `""` rather than `None`.

Hmm... so, checking for both `None` or an empty string here, then. Seems prudent to still check for `None` since that's what Django claims is possible in the documentation.

> Also, this loses extra information like the charset in `text/plain; charset=UTF-8`, which gets moved to the `user_file.content_type_extra` dict.

True. That's information we've been losing already. So, perhaps a follow-up issue for using that information as well? Currently, `LocalUploadBackend` doesn't use the `content_type` value when doing `upload_message_file`, while the `S3UploadBackend` does.

Reply:
Opened  #22682.